Terminal.Gui/View/IDesignable.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace Terminal.Gui;
+
+/// <summary>
+///     Interface declaring common functionality useful for designer implementations.
+/// </summary>
+public interface IDesignable
+{
+    /// <summary>
+    ///     Causes the View to enable design-time mode. This typically means that the view will load demo data and
+    ///     be configured to allow for design-time manipulation.
+    /// </summary>
+    /// <param name="context">Optional arbitrary, View-specific, context.</param>
+    /// <typeparam name="TContext">A non-null type for <paramref name="context"/>.</typeparam>
+    /// <returns><see langword="true"/> if the view successfully loaded demo data.</returns>
+    public bool EnableForDesign<TContext> (in TContext context) where TContext : notnull => EnableForDesign ();
```

Comment:
You know...

Can't believe this didn't occur to me while we were talking about it before.

`ref readonly` instead of `in` (yay c# 12!) might actually be the simplest answer here.

The difference is `ref readonly` is mandatory ref without a defensive copy at the callsite, which results in the behavior being the same for reference and value types.

It also carries the restrictions of not modifying the reference, as the `in` keyword does.

It requires no code change at the call sites, whether they currently include the `in` or not. In the former case, there's no change at all. In the latter case, the compiler emits a warning to notify you that the semantics are explicitly ref-always, rather than the C# default of value-unless-told-otherwise. Adding `in` or `ref` at the callsite makes the warning go away, as was the intent of the warning by design.

I think we can then get rid of the notnull, though I'll have to double-check that rule.

But I'm more than happy to tackle that and test it out in a separate branch if you'd rather get this one wrapped up for now.

Reply:
I changed it to `ref readonly`. Left `notnul` in.